Dispatcher/UI/Search/SearchScreenViewController.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  SearchScreenViewController.swift
+//  Dispatcher
+//
+//  Created by Ofek Cohen on 02/05/2023.
+//
+
+import UIKit
+
+class SearchScreenViewController: UIViewController {
+
+    @IBOutlet weak var searchTableView: UITableView!
+    @IBOutlet weak var clearButton: UIButton!
+    @IBOutlet weak var babkButton: UIButton!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    let searchScreenViewModel = SearchViewModel()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        customSearchBar()
+       
+        searchTableView.dataSource = self
+        searchTableView.delegate = self
+        searchTableView.register(UINib(nibName: AppConstants.latestSearchCellNibName, bundle: nil), forCellReuseIdentifier: AppConstants.latestSearchesCellIdentifier)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        searchScreenViewModel.fetchLatestSearchs()
+        searchTableView.reloadData()
+    }
+    
+    private func customSearchBar() {
+        let searchTextField = searchBar.searchTextField
+        searchBar.setImage(UIImage(named: "exit"), for: .clear, state: .normal)
+        
+        searchTextField.leftView = nil
+        searchTextField.backgroundColor = .white
+
+    }
+    
+    
+    @IBAction func backButtonPressed(_ sender: UIButton) {
+        self.dismiss(animated: true)
+    }
+    
+    @IBAction func clearButtonPressed(_ sender: UIButton) {
+        searchScreenViewModel.removeAllSearches()
+        searchTableView.reloadData()
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if (segue.identifier == AppConstants.fromLatestSearchToResults) {
+            let searchResultsVC = segue.destination as? SearchResultsScreenViewController
+            searchResultsVC?.searchKeyWords = searchBar.text ?? ""
+            searchResultsVC?.searchViewModel = searchScreenViewModel
+        }
+    }
+}
+
+
+extension SearchScreenViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchScreenViewModel.latestSearches.count
+        
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let currentSearchIndex = (searchScreenViewModel.latestSearches.count - indexPath.row) - 1
+        let search = searchScreenViewModel.latestSearches[currentSearchIndex]
+        
+        if let cell = (tableView.dequeueReusableCell(withIdentifier: AppConstants.latestSearchesCellIdentifier, for: indexPath) as? LatestSearchCell) {
```

Comment:
If we can't register the custom cell that we want we should return a regular UITableViewCell() (like you did in the homepage)